docs/advanced/settings.md

```diff
@@ -503,6 +503,61 @@ This setting can only be used if the `upgrade-checker-enabled` setting is set to
 https://your.upgrade.checker-url/v99/checkupgrade
 ```
 
+### `upgrade-config`
+
+**Definition**: Upgrade-related configuration.
+
+**Default value**: `{"imagePreloadOption":{"strategy":{"type":"sequential"}}}`
+
+**Example**:
+
+```json
+{
+  "imagePreloadOption": {
+    "strategy": {
+      "type": "parallel",
+      "concurrency": 2
+    }
+  }
+}
+```
+
+#### `imagePreloadOption`
+
+**Definition**: Options for the image-preloading phase.
+
+##### `strategy`
+
+**Definition**: Image preload strategy.
+
+###### `type`
+
+**Definition**: Type of the image preload strategy.
+
+There are three available types:
+
+- `skip`: Nodes will not preload images
+- `sequential`: Nodes will preload images one by one. This is the default.
+- `parallel`: Nodes will preload images in batches. This can be fine-tuned by the `concurrency` option.
+
+:::important
+
+Please do not use `skip` if your cluster is air-gapped. Harvester upgrade will fail if no new images are preloaded to the nodes or downloadable from any registry.
+
+:::
+
+###### `concurrency`
+
+**Definition**: Batch size of image preloading in terms of node count.
+
+Customize the number of nodes that can start preloading images simultaneously. The default value is `0`, which is equivalent to setting it with the cluster's node counts. One benefit of setting it to `0` is it dynamically follows the scale of the cluster. Values higher than the cluster's node counts are treated as `0`; values lower than 0 are invalid and will be rejected by Harvester.
+
+:::note
+
+The `concurrency` option only takes effect when the image-preload strategy `type` is set to `parallel`.
+
```

Comment:
Add another:

Harvester deploys a `cluster-repo` service on the cluster to serve as a http server for nodes to preload the image. When `concurrency` is set,  certain number of nodes will download the Harvester ISO from this `cluster-repo` in parallel,  following factors need to be considered:
(1) The Harvester management network speed.
(2) The writing speed of the default disk for Longhorn.

When your cluster has 20+ nodes, suggest to set the `concurrency` as `5`.


note: Before we have tested this on a at least medium sized cluster, it is better to suggest a conservative number, rather than 0.

Reply:
If we don't have data from experiments about the appropriate concurrency value, IMO we should not publish any concrete number in the documentation. Maybe just add a "this is an experimental function" warning?